pkg/network/tracer/tracer_usm_linux_test.go

```diff
@@ -285,43 +286,59 @@ func testHTTPSLibrary(t *testing.T, fetchCmd []string, prefetchLibs []string) {
 	requestCmd := exec.Command(cmd[0], cmd[1:]...)
 	out, err := requestCmd.CombinedOutput()
 	require.NoErrorf(t, err, "failed to issue request via %s: %s\n%s", fetchCmd, err, string(out))
+	fetchPid := uint32(requestCmd.Process.Pid)
+	t.Logf("%s pid %d", cmd[0], fetchPid)
 
+	var allConnections []network.ConnectionStats
+	var httpKey http.Key
 	require.Eventuallyf(t, func() bool {
 		payload := getConnections(t, tr)
+		allConnections = append(allConnections, payload.Conns...)
 		for key, stats := range payload.HTTP {
+			if key.Path.Content != "/200/foobar" {
+				continue
+			}
 			req, exists := stats.Data[200]
 			if !exists {
-				continue
+				errorLogs := fmt.Sprintf("http %# v stats %# v", krpretty.Formatter(key), krpretty.Formatter(stats.Data))
+				t.Logf("%s", errorLogs)
+				t.Fail()
 			}
 
 			statsTags := req.StaticTags
 			// debian 10 have curl binary linked with openssl and gnutls but use only openssl during tls query (there no runtime flag available)
 			// this make harder to map lib and tags, one set of tag should match but not both
 			if key.Path.Content == "/200/foobar" && (statsTags == network.ConnTagGnuTLS || statsTags == network.ConnTagOpenSSL) {
 				t.Logf("found tag 0x%x %s", statsTags, network.GetStaticTags(statsTags))
+				httpKey = key
+				return true
+			}
+			if key.Path.Content == "/200/foobar" {
+				t.Logf("HTTP stat didn't match criteria %v tags 0x%x\n", key, statsTags)
+			}
+		}
+		return false
+	}, 15*time.Second, 5*time.Second, "couldn't find USM HTTPS stats")
+
+	// socket filter is not supported on fentry tracer, so can't classify TLS protocol
 
-				// socket filter is not supported on fentry tracer
-				if fentryTracerEnabled {
-					// so we return early if the test was successful until now
+	if !fentryTracerEnabled {
```

Comment:
it is guaranteed that we are not running with fentry enabled
https://github.com/DataDog/datadog-agent/pull/16910

Reply:
shall we backport #16910 to 7.45 (and as well this one) ?

Reply:
IMHO no, we're merging only broken tests, flaky tests are not urgent